src/components/Events/EventFilteredGrid.jsx

```diff
@@ -0,0 +1,232 @@
+  const [location, setLocation] = useState('');
+  const [date, setDate] = useState('');
+  const [fuse, setFuse] = useState();
+
+  const getEvents = async () => {
+    try {
+      const eventsData = await Backend.get('/events');
+      setEvents(eventsData.data);
+      console.log(eventsData.data);
+      // setDates();
+      setLocations(getLocation(eventsData.data));
+      setDates(getDate(eventsData.data));
+      const options = { keys: ['name', 'date', 'location'], includeScore: true }; //use date and locaiton selected
+      setFuse(new Fuse(eventsData.data, options));
+    } catch (err) {
+      console.log(`Error getting events: `, err.message);
+    }
+  };
+
+  const getLocation = data => {
+    let location = [];
+    for (let i in data) {
+      location.push(data[i].location);
+    }
+    console.log(location.length);
+    return location;
+  };
+
+  const getDate = data => {
+    let date = [];
+    for (let i in data) {
+      date.push(data[i].date.substring(0, 10));
+    }
+    console.log(date.length);
+    return date;
+  };
+
+  const handleCheckboxChange = id => {
+    const newCheckedItems = [...selectedEvents];
+    const index = newCheckedItems.indexOf(id);
+
+    if (index === -1) {
+      newCheckedItems.push(id);
+    } else {
+      newCheckedItems.splice(index, 1);
+    }
+
+    setSelectedEvents(newCheckedItems);
+    console.log(selectedEvents);
+  };
+
+  useEffect(() => {
+    getEvents();
+    console.log(locations);
+    // getEventId(eventId);
+  }, []);
+
+  const handleLocationChange = event => {
+    const selectedLocation = event.target.value;
+    console.log(selectedLocation);
+    setLocation(selectedLocation);
+  };
+
+  const handleDateChange = event => {
+    const selectedDate = event.target.value;
+    console.log(selectedDate);
+    setDate(selectedDate);
+  };
+
+  const getLocationOptions = () => {
+    return locations.map((location, index) => (
+      <option key={location + index.toString()} value={location}>
+        {location}
+      </option>
+    ));
+  };
+
+  const getDateOptions = () => {
+    return dates.map((date, index) => (
+      <option key={date + index.toString()} value={date}>
+        {date}
+      </option>
+    ));
+  };
+
+  useEffect(() => {
+    if (!fuse) {
+      return;
+    }
+    console.log(name);
+    let ands = [];
+    if (name) ands.push({ name: name });
+    if (location) ands.push({ location: location });
+    if (date) ands.push({ date: date });
+
+    let result;
+    if (ands.length > 0) {
+      const fuseResult = fuse.search({ $and: ands });
+      console.log(fuseResult);
+      // If we want to filter by score:
+      // result = fuseResult.filter(item => item.score <= 0.5).map(item => item.item);
+      result = fuseResult.map(item => item.item);
+    } else result = events;
+    console.log(result);
+    // result.map((e) -> e.)
+    setDisplayEvents(result);
+  }, [name, location, date, fuse]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'events'. Either include it or remove the dependency array.